src/tools/adig.c

```diff
@@ -251,6 +262,80 @@ static ares_bool_t read_cmdline(int argc, const char * const *argv,
   return ARES_TRUE;
 }
 
+static ares_bool_t read_rcfile(adig_config_t *config)
+{
+  char                 configdir[256];
+  char                *configdir_xdg;
+  unsigned int         cdlen = 0;
+  unsigned int         cdsize = sizeof(configdir);
+  char                *homedir;
+
+#if defined(WIN32)
+  cdlen = (unsigned int) snprintf(configdir, cdsize, "%s/%s", getenv("APPDATA"), "c-ares");
+
+#elif defined(__APPLE__)
+  homedir = getenv("HOME");
+  if (homedir != NULL) {
+    cdlen = (unsigned int) snprintf(configdir, cdsize, "%s/%s/%s/%s", homedir, "Library", "Application Support", "c-ares");
+  }
+
+#else
+  configdir_xdg = getenv("XDG_CONFIG_HOME");
+
+  if (configdir_xdg == NULL) {
+    homedir = getenv("HOME");
+    if (homedir != NULL) {
+      cdlen = (unsigned int) snprintf(configdir, cdsize, "%s/%s", homedir, ".config");
+    }
+  }
+  else {
+    cdlen = (unsigned int) snprintf(configdir, cdsize, "%s", configdir_xdg);
+  }
+
+#endif
+
+  DEBUGF(fprintf(stderr, "read_cmdline() configdir: %s\n", configdir));
+
+  if (cdlen != 0 && cdlen < cdsize) {
```

Comment:
invert this if statement and just return out of the function early would probably make this more readable so you don't have an indented code block here.

Reply:
Fair point, I adjusted it and reduced some indentations.